streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -63,23 +65,30 @@
     private final Set<String> globalStoreNames = new HashSet<>();
     private final Map<TopicPartition, Long> checkpointableOffsets = new HashMap<>();
     private final StateRestoreListener stateRestoreListener;
+    private final int retries;
+    private final long retryBackoffMs;
 
-    public GlobalStateManagerImpl(final ProcessorTopology topology,
+    public GlobalStateManagerImpl(final LogContext logContext,
+                                  final ProcessorTopology topology,
                                   final Consumer<byte[], byte[]> globalConsumer,
                                   final StateDirectory stateDirectory,
-                                  final StateRestoreListener stateRestoreListener) {
+                                  final StateRestoreListener stateRestoreListener,
+                                  final StreamsConfig config) {
```

Comment:
I'd prefer to pass in the two config params here rather than the actual `StreamsConfig` we don't need the entire config.

Reply:
I prefer to keep parameter lists short... Seems to be a matter of taste... Note, we hit style check error in the past if parameter list are too long. Can change it if you insist on it of course. \cc @guozhangwang @bbejeck WDYT?

Reply:
I agree it is kind of a personal taste. Personally I prefer one `config` object than multiple ones.